src/osd/PrimaryLogPG.cc

```diff
@@ -6000,6 +6009,7 @@ int PrimaryLogPG::do_sparse_read(OpContext *ctx, OSDOp& osd_op) {
       r = rep_repair_primary_object(soid, ctx);
     }
     if (r < 0) {
+      dout(10) << " sparse_read failed r=" << r << " from object " << soid << dendl;
```

Comment:
Just to the record: I was looking how the new `objects_readv_sync()` is triggered. The answer is: just above `PGBackend::objects_readv_sync()` is called already:

```cpp
    bufferlist data_bl;
    r = pgbackend->objects_readv_sync(soid, m, op.flags, &data_bl);
    if (r == -EIO) {
      r = rep_repair_primary_object(soid, ctx);
    }
```
`ECSwitch` overloads the stub of `PGBackend`:

```cpp
   virtual int objects_readv_sync(
     const hobject_t &hoid,
     std::map<uint64_t, uint64_t>& m,
     uint32_t op_flags,
     ceph::buffer::list *bl) {
     return -EOPNOTSUPP;
   }
```